src/Bedrock.Framework.Experimental/Protocols/Http1RequestMessageWriter.cs

```diff
@@ -11,9 +13,33 @@ public class Http1RequestMessageWriter : IMessageWriter<HttpRequestMessage>
         private ReadOnlySpan<byte> Http11 => new byte[] { (byte)'H', (byte)'T', (byte)'T', (byte)'P', (byte)'/', (byte)'1', (byte)'.', (byte)'1' };
         private ReadOnlySpan<byte> NewLine => new byte[] { (byte)'\r', (byte)'\n' };
         private ReadOnlySpan<byte> Space => new byte[] { (byte)' ' };
+        private ReadOnlySpan<byte> Colon => new byte[] { (byte)':' };
+        private ReadOnlySpan<byte> Host => new byte[] { (byte)'H', (byte)'o', (byte)'s', (byte)'t' };
+
+        private const int DefaultHttpPort = 80;
+        private const int DefaultHttpsPort = 443;
+
+        private readonly string _host;
+        private readonly int _port;
+        private readonly byte[] _hostHeaderValueBytes;
+
+        public Http1RequestMessageWriter(string host, int port)
+        {
+            _host = host;
+            _port = port;
+
+            // Precalculate ASCII bytes for Host header
+            string hostHeader = _port != DefaultHttpPort && _port != DefaultHttpsPort
```

Comment:
Is this check necessary? I think it's valid to have port 80 or port 443 in the host header and this check feel clunky.